addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/handler/IhcHandler.java

```diff
@@ -0,0 +1,920 @@
+                    value = "Ready";
+                    break;
+                default:
+                    value = "Unknown state: " + state;
+            }
+
+            updateState(new ChannelUID(getThing().getUID(), CHANNEL_CONTROLLER_STATE), new StringType(value));
+        } catch (IhcExecption e) {
+            logger.warn("Controller state information fetch failed, reason:  {}", e.getMessage());
+        }
+    }
+
+    private void updateControllerProperties() {
+        try {
+            WSSystemInfo systemInfo = ihc.getSystemInfo();
+            logger.debug("Controller information: {}", systemInfo);
+            WSProjectInfo projectInfo = ihc.getProjectInfo();
+            logger.debug("Project information: {}", projectInfo);
+
+            Map<String, String> properties = editProperties();
+            properties.put(PROPERTY_MANUFACTURER, systemInfo.getBrand());
+            properties.put(PROPERTY_SERIALNUMBER, systemInfo.getSerialNumber());
+            properties.put(PROPERTY_SW_VERSION, systemInfo.getVersion());
+            properties.put(PROPERTY_FW_VERSION, systemInfo.getHwRevision());
+            properties.put(PROPERTY_APP_WITHOUT_VIEWER, Boolean.toString(systemInfo.getApplicationIsWithoutViewer()));
+            properties.put(PROPERTY_SW_DATE,
+                    systemInfo.getSwDate().withZoneSameInstant(ZoneId.systemDefault()).toString());
+            properties.put(PROPERTY_PRODUCTION_DATE, systemInfo.getProductionDate());
+            if (!systemInfo.getDatalineVersion().isEmpty()) {
+                properties.put(PROPERTY_DATALINE_VERSION, systemInfo.getDatalineVersion());
+            }
+            if (!systemInfo.getRfModuleSerialNumber().isEmpty()) {
+                properties.put(PROPERTY_RF_MODULE_SERIALNUMBER, systemInfo.getRfModuleSerialNumber());
+            }
+            if (!systemInfo.getRfModuleSoftwareVersion().isEmpty()) {
+                properties.put(PROPERTY_RF_MODULE_VERSION, systemInfo.getRfModuleSoftwareVersion());
+            }
+            properties.put(PROPERTY_PROJECT_DATE,
+                    projectInfo.getLastmodified().getAsLocalDateTime().atZone(ZoneId.systemDefault()).toString());
+            properties.put(PROPERTY_PROJECT_NUMBER, projectInfo.getProjectNumber());
+            updateProperties(properties);
+        } catch (IhcExecption e) {
+            logger.warn("Controller information fetch failed, reason:  {}", e.getMessage());
+        }
+    }
+
+    private void updateControllerInformationChannels() {
+        try {
+            WSSystemInfo systemInfo = ihc.getSystemInfo();
+            logger.debug("Controller information: {}", systemInfo);
+
+            updateState(new ChannelUID(getThing().getUID(), CHANNEL_CONTROLLER_UPTIME),
+                    new DecimalType((double) systemInfo.getUptime() / 1000));
+        } catch (IhcExecption e) {
+            logger.warn("Controller uptime information fetch failed, reason:  {}", e.getMessage());
```

Comment:
You lose the stacktrace here, could it contain something useful?

Reply:
Done